src/liger_kernel/ops/fused_linear_cross_entropy.py

```diff
@@ -109,7 +119,9 @@ def forward(ctx, _input, weight, target, bias=None, ignore_index=-100):
             # thus grad_input[start_idx: end_idx] should be of shape: chunk_size x H
             # additionally, since we are chunking the inputs, observe that the loss and gradients are calculated only
             # on `n_non_ignore` tokens. However, the gradient of the input should be calculated for all tokens.
-            # Thus, we need an additional scaling factor of (n_non_ignore/total_n_non_ignore) to scale the gradients.
+            # Thus, we need an additional scaling factor of (n_non_ignore/total_n_non_ignore) to scale the loss and gradients.
+
+            loss_1d[start_idx:end_idx] = loss_1d_slice * n_non_ignore
```

Comment:
for "mean", i think we need to scale by `n_non_ignore / total_n_non_ignore`?
for others, do we need to scale? no?